drivers/dac/dac_ad559x.c

```diff
@@ -88,6 +89,19 @@ static int dac_ad559x_init(const struct device *dev)
 		return -ENODEV;
 	}
 
+	if (config->range_double) {
+		uint16_t tmp;
+
+		ret = mfd_ad559x_read_reg(config->mfd_dev, AD559X_REG_GEN_CTRL, 0, &tmp);
+		if (ret < 0) {
+			return ret;
+		}
+		ret = mfd_ad559x_write_reg(config->mfd_dev, AD559X_REG_GEN_CTRL, tmp);
```

Comment:
How is it supposed to work? You do a read and write the same register without any modification.